`Test-VcfPasswordManagementPrereq` assumes modules installed via PowerShellGet

### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/blob/main/CODE_OF_CONDUCT.md).
- [ ] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [ ] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### VMware Cloud Foundation

v4.5

### PowerShell Version

PowerShell Core 7.2.8

### PowerCLI Version

13.1.0.21624340

### Module Version

1.1.0.1008

### PowerValidatedSolutions Version

2.3.0.1007

### PowerVCF Version

2.3.0.1004

### Guest Operating System

Windows Server 2019

### Environment Details

Modules have been installed to the system via Windows PowerShell 5.1, to the all users scope. They are installed to "C:\Program Files\WindowsPowerShell\Modules"

PowerShell Core also reads in modules from this location, the value of $env:PSModulePath = "C:\Users\user\Documents\PowerShell\Modules;C:\Program Files\PowerShell\Modules;c:\program files\powershell\7\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules"

### Description

Test-VcfPasswordManagementPrereq reports [ERROR] PowerShell Module: VMware.PowerCLI 13.0.0 minimum required version is not installed

Modules have been imported in PS 7.2.8 admin session:
Import-Module -Name VMware.PowerCLI
Import-Module -Name VMware.vSphere.SsoAdmin
Import-Module -Name PowerVCF
Import-Module -Name PowerValidatedSolutions
Import-Module -Name VMware.CloudFoundation.PasswordManagement

### Error or Debug Output

[ERROR] PowerShell Module: VMware.PowerCLI 13.0.0 minimum required version is not installed.

Get-Module -ListAvailable reports modules as available (see attached).

[modules under ps core 7.2.8.log](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/files/11861581/modules.under.ps.core.7.2.8.log)


### Expected Behavior

Test-VcfPasswordManagementPrereq should return output as below:

 [06-26-2023_13:03:30] [INFO] PowerShell Module: VMware.PowerCLI 13.1.0.21624340 is installed and supports the minimum required version.
 [06-26-2023_13:03:33] [INFO] PowerShell Module: VMware.vSphere.SsoAdmin 1.3.9 is installed and supports the minimum required version.
 [06-26-2023_13:03:36] [INFO] PowerShell Module: PowerVCF 2.3.0.1004 is installed and supports the minimum required version.
 [06-26-2023_13:03:39] [INFO] PowerShell Module: PowerValidatedSolutions 2.3.0.1007 is installed and supports the minimum required version.

### Actual Behavior

Test-VcfPasswordManagementPrereq output:

[06-26-2023_13:19:33] [ERROR] PowerShell Module: VMware.PowerCLI 13.0.0 minimum required version is not installed.

### Steps to Reproduce

- Install modules in Windows PowerShell 5.1:
Set-PSRepository -Name PSGallery -InstallationPolicy Trusted
Install-Module -Name VMware.PowerCLI -MinimumVersion 13.0.0
Install-Module -Name VMware.vSphere.SsoAdmin -MinimumVersion 1.3.9
Install-Module -Name PowerVCF -MinimumVersion 2.3.0
Install-Module -Name PowerValidatedSolutions -MinimumVersion 2.3.0
Install-Module -Name VMware.CloudFoundation.PasswordManagement

Open PowerShell Core 7.2.8
Run Test-VcfPasswordManagementPrereq

### Log Fragments and Files

_No response_

### Screenshots

_No response_

### References

Test-VcfPasswordManagementPrereq uses the PowerShell command [Get-InstalledModule](https://learn.microsoft.com/en-us/powershell/module/powershellget/get-installedmodule?view=powershellget-2.x), which gets a list of modules on the computer that were installed by PowerShellGet.

This command unfortunately assumes that the modules have been installed via PowerShellGet.

We have two systems where modules are not installed via PowerShellGet, and both show similar issues with Test-VcfPasswordManagementPrereq
1. This system that we use for testing has both Windows PowerShell 5.1 and PowerShell Core
2. Linux PowerShell Core systems that we use for automation with GitLab runners, and have restricted internet access, so modules are copied into paths referenceable by $env:PSModulePath

An alternative that might cover all these scenarion could be:
Get-Module or Get-Module -ListAvailable, similar to https://stackoverflow.com/questions/28740320/how-do-i-check-if-a-powershell-module-is-installed

Our documentation currently assumes that the module and dependencies are installed from the PowerShell Gallery; however, there is an opportunity to enhance the documentation to cover a disconnected / dark-site installation example.

<details><summary>Successful on VMware Photon OS 4.0 with PowerShell 7.2.7</summary>

```powershell
vmware@jbx [ ~ ]$ pwsh
PowerShell 7.2.7
Copyright (c) Microsoft Corporation.

https://aka.ms/powershell
Type 'help' to get help.

   A new PowerShell stable release is available: v7.3.5
   Upgrade now, or check out the release page at:
     https://aka.ms/PowerShell-Release?tag=v7.3.5

PS /home/vmware> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.7
PSEdition                      Core
GitCommitId                    7.2.7
OS                             Linux 5.10.109-2.ph4-esx #1-photon SMP Sat May 14 02:59:09 UTC 2022
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS /home/vmware>
PS /home/vmware/.local/share/powershell/Modules> Get-Module

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Manifest   7.0.0.0               Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty…}
Manifest   7.0.0.0               Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object…}
Script     1.4.7                 PackageManagement                   {Find-Package, Find-PackageProvider, Get-Package, Get-PackageProvider…}
Script     2.2.5                 PowerShellGet                       {Find-Command, Find-DscResource, Find-Module, Find-RoleCapability…}
Script     2.1.0                 PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSReadLineKeyHandler, Set-PSReadLineKeyHandler…}
Script     12.6.0.19…            VMware.CloudServices                {Connect-Vcs, Disconnect-Vcs, Get-VcsOrganizationRole, Get-VcsService…}
Script     8.0.0.216…            VMware.DeployAutomation             {Add-CustomCertificate, Add-DeployRule, Add-ProxyServer, Add-ScriptBundle…}
Script     8.0.0.216…            VMware.ImageBuilder                 {Add-EsxSoftwareDepot, Add-EsxSoftwarePackage, Compare-EsxImageProfile, Export-EsxImageProfile…}
Script     12.6.0.19…            VMware.PowerCLI.Sdk
Script     12.6.0.19…            VMware.PowerCLI.Sdk.Types
Script     12.6.0.19…            VMware.PowerCLI.VCenter             {Add-VITrustedCertificate, Get-VIApplianceService, Get-VIMachineCertificate, Get-VITrustedCertificate…}
Script     12.6.0.19…            VMware.PowerCLI.VCenter.Types.Appl…
Script     12.6.0.19…            VMware.PowerCLI.VCenter.Types.Cert…
Script     4.1.0.216…            VMware.Sdk.Nsx.Policy               {Connect-NsxServer, Connect-NsxVmcServer, Disconnect-NsxServer, Disconnect-NsxVmcServer…}
Script     1.0.1111.…            VMware.Sdk.Runtime                  {Add-ServerConfiguration, ConvertFrom-JsonX, Get-ServerConfiguration, Invoke-ApiClient…}
Script     8.7.0.216…            VMware.Sdk.Srm                      {Connect-SrmSdkServer, Disconnect-SrmSdkServer, Get-SrmOperation, Initialize-SrmAbrProtectionGroupSpec…}
Script     8.7.0.216…            VMware.Sdk.Vr                       {Connect-VrServer, Disconnect-VrServer, Get-VrOperation, Initialize-VrConfigureReplicationSpec…}
Manifest   8.0.1111.…            VMware.Sdk.vSphere
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance        {Initialize-HealthCheckSettingsUpdateSpec, Initialize-LocalAccountsConfig, Initialize-LocalAccountsCreateR…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Access {Initialize-AccessConsolecliSetRequestBody, Initialize-AccessDcuiSetRequestBody, Initialize-AccessShellShe…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Health {Invoke-GetHealthApplmgmt, Invoke-GetHealthDatabase, Invoke-GetHealthDatabaseStorage, Invoke-GetHealthLoad…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Infra… {Initialize-InfraprofileConfigsImportProfileSpec, Initialize-InfraprofileConfigsProfilesSpec, Invoke-Expor…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Local… {Initialize-LocalAccountsPolicyInfo, Invoke-GetLocalAccountsGlobalPolicy, Invoke-SetLocalAccountsGlobalPol…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Loggi… {Initialize-LoggingForwardingConfig, Initialize-LoggingForwardingSetRequestBody, Initialize-LoggingForward…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Netwo… {Initialize-NetworkingDnsDomainsAddRequestBody, Initialize-NetworkingDnsDomainsSetRequestBody, Initialize-…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Recov… {Initialize-RecoveryBackupBackupRequest, Initialize-RecoveryBackupJobBackupRequest, Initialize-RecoveryBac…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Suppo… Invoke-GetSupportBundleComponents
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.System {Initialize-SystemSecurityGlobalFipsUpdateSpec, Initialize-SystemTimeTimezoneSetRequestBody, Invoke-GetSys…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Update {Initialize-UpdatePendingInstallRequestBody, Initialize-UpdatePendingStageAndInstallRequestBody, Initializ…
Script     8.0.1111.…            VMware.Sdk.vSphere.Cis              {Initialize-TasksFilterSpec, Initialize-TasksGetSpec, Initialize-TasksListRequestBody, Invoke-CancelTask…}
Script     8.0.1111.…            VMware.Sdk.vSphere.Cis.Tagging      {Initialize-StdDynamicID, Initialize-TaggingCategoryAddToUsedByRequestBody, Initialize-TaggingCategoryCrea…
Script     8.0.1111.…            VMware.Sdk.vSphere.Content          {Initialize-ConfigurationModel, Initialize-LibraryFindSpec, Initialize-LibraryModel, Initialize-LibraryOpt…
Script     8.0.1111.…            VMware.Sdk.vSphere.ContentLibrary   {Initialize-LibraryItemCertificateVerificationInfo, Initialize-LibraryItemDestinationSpec, Initialize-Libr…
Script     8.0.1111.…            VMware.Sdk.vSphere.Esx.Hcl          {Initialize-HclHostsCompatibilityReportSpec, Invoke-CreateHostCompatibilityReportAsync, Invoke-DownloadHcl…
Script     8.0.1111.…            VMware.Sdk.vSphere.Esx.Hosts        {Initialize-HostsSoftwareHostCredentials, Invoke-GetSoftware, Invoke-ListHostSoftwareInstalledComponents}
Script     8.0.1111.…            VMware.Sdk.vSphere.Esx.Settings     {Initialize-SettingsAddOnSpec, Initialize-SettingsBaseImageSpec, Initialize-SettingsClustersConfigurationA…
Script     8.0.1111.…            VMware.Sdk.vSphere.VAPI.Metadata    {Initialize-MetadataCliCommandGetRequestBody, Initialize-MetadataCliCommandIdentity, Initialize-MetadataCl…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter          {Initialize-DatacenterCreateSpec, Initialize-HostCreateSpec, Initialize-ResourcePoolCreateSpec, Initialize…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Authent… Invoke-CreateAuthenticationToken
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Authori… {Initialize-AuthorizationPrivilegeChecksFilterSpec, Initialize-AuthorizationPrivilegeChecksListRequestBody…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.CertMan… {Initialize-CertificateManagementVcenterSigningCertificateRefreshRequestBody, Initialize-CertificateManage…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Consump… {Initialize-ConsumptionDomainsZonesCreateSpec, Invoke-AddZoneAssociations, Invoke-CreateConsumptionDomains…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Content  {Initialize-ContentRegistriesHarborCreateSpec, Initialize-ContentRegistriesHarborGarbageCollection, Initia…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Datasto… Invoke-GetDatastoreDefaultPolicy
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Deploym… {Initialize-DeploymentHistoryMigrationSpec, Initialize-DeploymentImportHistoryCreateSpec, Initialize-Deplo…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Guest    {Initialize-GuestCustomizationSpecsCreateSpec, Initialize-GuestCustomizationSpecsExportRequestBody, Initia…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.HVC      {Initialize-HvcLinksCreateSpec, Initialize-HvcLinksCredentials, Initialize-HvcLinksDeleteWithCredentialsRe…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Identity {Initialize-IdentityProvidersActiveDirectoryOverLdap, Initialize-IdentityProvidersCreateSpec, Initialize-I…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Invento… {Invoke-GetInventoryDatastore, Invoke-GetInventoryNetwork}
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.ISO      {Initialize-IsoImageMountRequestBody, Initialize-IsoImageUnmountRequestBody, Invoke-MountIsoImage, Invoke-…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.LCM      {Initialize-LcmDiscoveryAssociatedProductsCreateSpec, Initialize-LcmDiscoveryAssociatedProductsUpdateSpec,…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Namespa… {Initialize-NamespaceManagementClustersEnableSpec, Initialize-NamespaceManagementClustersImageRegistry, In…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Namespa… {Initialize-NamespacesAccessCreateSpec, Initialize-NamespacesAccessSetSpec, Initialize-NamespacesInstances…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.OVF      {Initialize-OvfLibraryItemCreateRequestBody, Initialize-OvfLibraryItemCreateSpec, Initialize-OvfLibraryIte…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Services {Initialize-ServicesServiceUpdateSpec, Invoke-GetService, Invoke-GetServices, Invoke-RestartService…}
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Storage  {Initialize-StoragePoliciesCheckCompatibilityRequestBody, Invoke-CheckCompatibilityPolicyPolicies, Invoke-…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.SystemC… {Initialize-DeploymentRemotePscSpec, Initialize-SystemConfigDeploymentTypeReconfigureSpec, Invoke-GetSyste…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Tagging  Invoke-ListAssociations
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Topology {Invoke-GetNodeTopology, Invoke-ListTopologyNodes, Invoke-ListTopologyReplicationStatus}
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Trusted… {Initialize-TrustedInfrastructureAttestationServicesCreateSpec, Initialize-TrustedInfrastructureAttestatio…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.VCHA     {Initialize-VchaClusterActiveGetRequestBody, Initialize-VchaClusterActiveSpec, Initialize-VchaClusterDeplo…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Vm       {Initialize-GuestAdapterMapping, Initialize-GuestCloudConfiguration, Initialize-GuestCloudinitConfiguratio…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.VmTempl… {Initialize-VmTemplateLibraryItemsCheckOutsCheckInSpec, Initialize-VmTemplateLibraryItemsCheckOutsCheckOut…
Script     8.0.1111.…            VMware.Sdk.vSphere.vStats           {Initialize-AcqSpecsCounterSpec, Initialize-AcqSpecsCreateSpec, Initialize-AcqSpecsUpdateSpec, Initialize-…
Script     8.0.1111.…            VMware.Sdk.vSphereRuntime           {Convert-InputStructure, Convert-OutputBody, Format-Body, Format-Headers…}
Script     8.1.0.216…            VMware.Vim
Script     13.1.0.21…            VMware.VimAutomation.Cis.Core       {Connect-CisServer, Disconnect-CisServer, Get-CisService}
Script     13.1.0.21…            VMware.VimAutomation.Cloud          {Add-CIDatastore, Connect-CIServer, Disconnect-CIServer, Get-Catalog…}
Script     13.1.0.21…            VMware.VimAutomation.Common         {Get-PowerCLIContext, Get-Task, New-OAuthSecurityContext, Stop-Task…}
Script     13.1.0.21…            VMware.VimAutomation.Core           {Add-PassthroughDevice, Add-VirtualSwitchPhysicalNetworkAdapter, Add-VMHost, Add-VMHostNtpServer…}
Script     13.0.0.20…            VMware.VimAutomation.Hcx            {Connect-HCXServer, Disconnect-HCXServer, Get-HCXAppliance, Get-HCXComputeProfile…}
Script     13.1.0.21…            VMware.VimAutomation.HorizonView    {Connect-HVServer, Disconnect-HVServer}
Script     12.0.0.15…            VMware.VimAutomation.License        Get-LicenseDataManager
Script     13.1.0.21…            VMware.VimAutomation.Nsxt           {Connect-NsxtServer, Disconnect-NsxtServer, Get-NsxtGlobalManagerService, Get-NsxtPolicyService…}
Script     13.1.0.21…            VMware.VimAutomation.Sdk            {Get-ErrorReport, EnableParameterCompleters, Get-InstallPath, Get-PSVersion}
Script     13.1.0.21…            VMware.VimAutomation.Security       {Add-AttestationServiceInfo, Add-KeyProviderServiceInfo, Add-TrustAuthorityKeyProviderServer, Add-TrustAut…
Script     12.7.0.20…            VMware.VimAutomation.Srm            {Connect-SrmServer, Disconnect-SrmServer}
Script     13.1.0.21…            VMware.VimAutomation.Storage        {Add-EntityDefaultKeyProvider, Add-KeyManagementServer, Add-VsanFileServiceOvf, Add-VsanObjectToRepairQueu…
Script     1.6.0.0               VMware.VimAutomation.StorageUtility Update-VmfsDatastore
Script     13.1.0.21…            VMware.VimAutomation.Vds            {Add-VDSwitchPhysicalNetworkAdapter, Add-VDSwitchVMHost, Export-VDPortGroup, Export-VDSwitch…}
Script     13.0.0.20…            VMware.VimAutomation.Vmc            {Add-VmcSddcHost, Connect-Vmc, Disconnect-Vmc, Get-AwsAccount…}
Script     13.1.0.21…            VMware.VimAutomation.vROps          {Connect-OMServer, Disconnect-OMServer, Get-OMAlert, Get-OMAlertDefinition…}
Script     12.4.0.18…            VMware.VimAutomation.WorkloadManag… {Disable-WMCluster, Enable-WMCluster, Get-WMCluster, Get-WMNamespace…}
Script     12.7.0.20…            VMware.VumAutomation                {Add-EntityBaseline, Copy-Patch, Get-Baseline, Get-Compliance…}

PS /home/vmware/.local/share/powershell/Modules> Import-Module ./PowerVCF/2.3.0.1004/PowerVCF.psd1
PS /home/vmware/.local/share/powershell/Modules> Import-Module ./PowerValidatedSolutions/2.4.0.1008/PowerValidatedSolutions.psd1
PS /home/vmware/.local/share/powershell/Modules> Import-Module ./VMware.vSphere.SsoAdmin/1.3.9/VMware.vSphere.SsoAdmin.psd1
PS /home/vmware/.local/share/powershell/Modules> Import-Module ./VMware.CloudFoundation.PasswordManagement/1.2.0.1005/VMware.CloudFoundation.PasswordManagement.psd1
PS /home/vmware/.local/share/powershell/Modules> Import-Module ./VMware.PowerCLI/13.1.0.21624340/VMware.PowerCLI.psd1

PS /home/vmware/.local/share/powershell/Modules> Get-Module

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Manifest   7.0.0.0               Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty…}
Manifest   7.0.0.0               Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object…}
Script     1.4.7                 PackageManagement                   {Find-Package, Find-PackageProvider, Get-Package, Get-PackageProvider…}
Script     2.2.5                 PowerShellGet                       {Find-Command, Find-DscResource, Find-Module, Find-RoleCapability…}
Script     2.4.0.1008            PowerValidatedSolutions             {Add-AntiAffinityRule, Add-CEPTrustedCertificate, Add-CEPvCenterServer, Add-CertToNsxCertificateStore…}
Script     2.3.0.1004            PowerVCF                            {Add-VCFApplicationVirtualNetwork, Add-VCFNetworkIPPool, Connect-CloudBuilder, Debug-CatchWriter…}
Script     2.1.0                 PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSReadLineKeyHandler, Set-PSReadLineKeyHandler…}
Script     1.2.0.1005            VMware.CloudFoundation.PasswordMan… {Get-PasswordPolicyConfig, Get-PasswordPolicyDefault, Invoke-PasswordPolicyManager, Publish-EsxiPasswordPo…
Script     12.6.0.19…            VMware.CloudServices                {Connect-Vcs, Disconnect-Vcs, Get-VcsOrganizationRole, Get-VcsService…}
Script     8.0.0.216…            VMware.DeployAutomation             {Add-CustomCertificate, Add-DeployRule, Add-ProxyServer, Add-ScriptBundle…}
Script     8.0.0.216…            VMware.ImageBuilder                 {Add-EsxSoftwareDepot, Add-EsxSoftwarePackage, Compare-EsxImageProfile, Export-EsxImageProfile…}
Manifest   13.1.0.21…            VMware.PowerCLI
Script     12.6.0.19…            VMware.PowerCLI.Sdk
Script     12.6.0.19…            VMware.PowerCLI.Sdk.Types
Script     12.6.0.19…            VMware.PowerCLI.VCenter             {Add-VITrustedCertificate, Get-VIApplianceService, Get-VIMachineCertificate, Get-VITrustedCertificate…}
Script     12.6.0.19…            VMware.PowerCLI.VCenter.Types.Appl…
Script     12.6.0.19…            VMware.PowerCLI.VCenter.Types.Cert…
Script     4.1.0.216…            VMware.Sdk.Nsx.Policy               {Connect-NsxServer, Connect-NsxVmcServer, Disconnect-NsxServer, Disconnect-NsxVmcServer…}
Script     1.0.1111.…            VMware.Sdk.Runtime                  {Add-ServerConfiguration, ConvertFrom-JsonX, Get-ServerConfiguration, Invoke-ApiClient…}
Script     8.7.0.216…            VMware.Sdk.Srm                      {Connect-SrmSdkServer, Disconnect-SrmSdkServer, Get-SrmOperation, Initialize-SrmAbrProtectionGroupSpec…}
Script     8.7.0.216…            VMware.Sdk.Vr                       {Connect-VrServer, Disconnect-VrServer, Get-VrOperation, Initialize-VrConfigureReplicationSpec…}
Manifest   8.0.1111.…            VMware.Sdk.vSphere
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance        {Initialize-HealthCheckSettingsUpdateSpec, Initialize-LocalAccountsConfig, Initialize-LocalAccountsCreateR…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Access {Initialize-AccessConsolecliSetRequestBody, Initialize-AccessDcuiSetRequestBody, Initialize-AccessShellShe…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Health {Invoke-GetHealthApplmgmt, Invoke-GetHealthDatabase, Invoke-GetHealthDatabaseStorage, Invoke-GetHealthLoad…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Infra… {Initialize-InfraprofileConfigsImportProfileSpec, Initialize-InfraprofileConfigsProfilesSpec, Invoke-Expor…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Local… {Initialize-LocalAccountsPolicyInfo, Invoke-GetLocalAccountsGlobalPolicy, Invoke-SetLocalAccountsGlobalPol…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Loggi… {Initialize-LoggingForwardingConfig, Initialize-LoggingForwardingSetRequestBody, Initialize-LoggingForward…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Netwo… {Initialize-NetworkingDnsDomainsAddRequestBody, Initialize-NetworkingDnsDomainsSetRequestBody, Initialize-…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Recov… {Initialize-RecoveryBackupBackupRequest, Initialize-RecoveryBackupJobBackupRequest, Initialize-RecoveryBac…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Suppo… Invoke-GetSupportBundleComponents
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.System {Initialize-SystemSecurityGlobalFipsUpdateSpec, Initialize-SystemTimeTimezoneSetRequestBody, Invoke-GetSys…
Script     8.0.1111.…            VMware.Sdk.vSphere.Appliance.Update {Initialize-UpdatePendingInstallRequestBody, Initialize-UpdatePendingStageAndInstallRequestBody, Initializ…
Script     8.0.1111.…            VMware.Sdk.vSphere.Cis              {Initialize-TasksFilterSpec, Initialize-TasksGetSpec, Initialize-TasksListRequestBody, Invoke-CancelTask…}
Script     8.0.1111.…            VMware.Sdk.vSphere.Cis.Tagging      {Initialize-StdDynamicID, Initialize-TaggingCategoryAddToUsedByRequestBody, Initialize-TaggingCategoryCrea…
Script     8.0.1111.…            VMware.Sdk.vSphere.Content          {Initialize-ConfigurationModel, Initialize-LibraryFindSpec, Initialize-LibraryModel, Initialize-LibraryOpt…
Script     8.0.1111.…            VMware.Sdk.vSphere.ContentLibrary   {Initialize-LibraryItemCertificateVerificationInfo, Initialize-LibraryItemDestinationSpec, Initialize-Libr…
Script     8.0.1111.…            VMware.Sdk.vSphere.Esx.Hcl          {Initialize-HclHostsCompatibilityReportSpec, Invoke-CreateHostCompatibilityReportAsync, Invoke-DownloadHcl…
Script     8.0.1111.…            VMware.Sdk.vSphere.Esx.Hosts        {Initialize-HostsSoftwareHostCredentials, Invoke-GetSoftware, Invoke-ListHostSoftwareInstalledComponents}
Script     8.0.1111.…            VMware.Sdk.vSphere.Esx.Settings     {Initialize-SettingsAddOnSpec, Initialize-SettingsBaseImageSpec, Initialize-SettingsClustersConfigurationA…
Script     8.0.1111.…            VMware.Sdk.vSphere.VAPI.Metadata    {Initialize-MetadataCliCommandGetRequestBody, Initialize-MetadataCliCommandIdentity, Initialize-MetadataCl…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter          {Initialize-DatacenterCreateSpec, Initialize-HostCreateSpec, Initialize-ResourcePoolCreateSpec, Initialize…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Authent… Invoke-CreateAuthenticationToken
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Authori… {Initialize-AuthorizationPrivilegeChecksFilterSpec, Initialize-AuthorizationPrivilegeChecksListRequestBody…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.CertMan… {Initialize-CertificateManagementVcenterSigningCertificateRefreshRequestBody, Initialize-CertificateManage…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Consump… {Initialize-ConsumptionDomainsZonesCreateSpec, Invoke-AddZoneAssociations, Invoke-CreateConsumptionDomains…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Content  {Initialize-ContentRegistriesHarborCreateSpec, Initialize-ContentRegistriesHarborGarbageCollection, Initia…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Datasto… Invoke-GetDatastoreDefaultPolicy
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Deploym… {Initialize-DeploymentHistoryMigrationSpec, Initialize-DeploymentImportHistoryCreateSpec, Initialize-Deplo…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Guest    {Initialize-GuestCustomizationSpecsCreateSpec, Initialize-GuestCustomizationSpecsExportRequestBody, Initia…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.HVC      {Initialize-HvcLinksCreateSpec, Initialize-HvcLinksCredentials, Initialize-HvcLinksDeleteWithCredentialsRe…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Identity {Initialize-IdentityProvidersActiveDirectoryOverLdap, Initialize-IdentityProvidersCreateSpec, Initialize-I…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Invento… {Invoke-GetInventoryDatastore, Invoke-GetInventoryNetwork}
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.ISO      {Initialize-IsoImageMountRequestBody, Initialize-IsoImageUnmountRequestBody, Invoke-MountIsoImage, Invoke-…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.LCM      {Initialize-LcmDiscoveryAssociatedProductsCreateSpec, Initialize-LcmDiscoveryAssociatedProductsUpdateSpec,…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Namespa… {Initialize-NamespaceManagementClustersEnableSpec, Initialize-NamespaceManagementClustersImageRegistry, In…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Namespa… {Initialize-NamespacesAccessCreateSpec, Initialize-NamespacesAccessSetSpec, Initialize-NamespacesInstances…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.OVF      {Initialize-OvfLibraryItemCreateRequestBody, Initialize-OvfLibraryItemCreateSpec, Initialize-OvfLibraryIte…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Services {Initialize-ServicesServiceUpdateSpec, Invoke-GetService, Invoke-GetServices, Invoke-RestartService…}
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Storage  {Initialize-StoragePoliciesCheckCompatibilityRequestBody, Invoke-CheckCompatibilityPolicyPolicies, Invoke-…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.SystemC… {Initialize-DeploymentRemotePscSpec, Initialize-SystemConfigDeploymentTypeReconfigureSpec, Invoke-GetSyste…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Tagging  Invoke-ListAssociations
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Topology {Invoke-GetNodeTopology, Invoke-ListTopologyNodes, Invoke-ListTopologyReplicationStatus}
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Trusted… {Initialize-TrustedInfrastructureAttestationServicesCreateSpec, Initialize-TrustedInfrastructureAttestatio…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.VCHA     {Initialize-VchaClusterActiveGetRequestBody, Initialize-VchaClusterActiveSpec, Initialize-VchaClusterDeplo…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.Vm       {Initialize-GuestAdapterMapping, Initialize-GuestCloudConfiguration, Initialize-GuestCloudinitConfiguratio…
Script     8.0.1111.…            VMware.Sdk.vSphere.vCenter.VmTempl… {Initialize-VmTemplateLibraryItemsCheckOutsCheckInSpec, Initialize-VmTemplateLibraryItemsCheckOutsCheckOut…
Script     8.0.1111.…            VMware.Sdk.vSphere.vStats           {Initialize-AcqSpecsCounterSpec, Initialize-AcqSpecsCreateSpec, Initialize-AcqSpecsUpdateSpec, Initialize-…
Script     8.0.1111.…            VMware.Sdk.vSphereRuntime           {Convert-InputStructure, Convert-OutputBody, Format-Body, Format-Headers…}
Script     8.1.0.216…            VMware.Vim
Script     13.1.0.21…            VMware.VimAutomation.Cis.Core       {Connect-CisServer, Disconnect-CisServer, Get-CisService}
Script     13.1.0.21…            VMware.VimAutomation.Cloud          {Add-CIDatastore, Connect-CIServer, Disconnect-CIServer, Get-Catalog…}
Script     13.1.0.21…            VMware.VimAutomation.Common         {Get-PowerCLIContext, Get-Task, New-OAuthSecurityContext, Stop-Task…}
Script     13.1.0.21…            VMware.VimAutomation.Core           {Add-PassthroughDevice, Add-VirtualSwitchPhysicalNetworkAdapter, Add-VMHost, Add-VMHostNtpServer…}
Script     13.0.0.20…            VMware.VimAutomation.Hcx            {Connect-HCXServer, Disconnect-HCXServer, Get-HCXAppliance, Get-HCXComputeProfile…}
Script     13.1.0.21…            VMware.VimAutomation.HorizonView    {Connect-HVServer, Disconnect-HVServer}
Script     12.0.0.15…            VMware.VimAutomation.License        Get-LicenseDataManager
Script     13.1.0.21…            VMware.VimAutomation.Nsxt           {Connect-NsxtServer, Disconnect-NsxtServer, Get-NsxtGlobalManagerService, Get-NsxtPolicyService…}
Script     13.1.0.21…            VMware.VimAutomation.Sdk            {Get-ErrorReport, EnableParameterCompleters, Get-InstallPath, Get-PSVersion}
Script     13.1.0.21…            VMware.VimAutomation.Security       {Add-AttestationServiceInfo, Add-KeyProviderServiceInfo, Add-TrustAuthorityKeyProviderServer, Add-TrustAut…
Script     12.7.0.20…            VMware.VimAutomation.Srm            {Connect-SrmServer, Disconnect-SrmServer}
Script     13.1.0.21…            VMware.VimAutomation.Storage        {Add-EntityDefaultKeyProvider, Add-KeyManagementServer, Add-VsanFileServiceOvf, Add-VsanObjectToRepairQueu…
Script     1.6.0.0               VMware.VimAutomation.StorageUtility Update-VmfsDatastore
Script     13.1.0.21…            VMware.VimAutomation.Vds            {Add-VDSwitchPhysicalNetworkAdapter, Add-VDSwitchVMHost, Export-VDPortGroup, Export-VDSwitch…}
Script     13.0.0.20…            VMware.VimAutomation.Vmc            {Add-VmcSddcHost, Connect-Vmc, Disconnect-Vmc, Get-AwsAccount…}
Script     13.1.0.21…            VMware.VimAutomation.vROps          {Connect-OMServer, Disconnect-OMServer, Get-OMAlert, Get-OMAlertDefinition…}
Script     12.4.0.18…            VMware.VimAutomation.WorkloadManag… {Disable-WMCluster, Enable-WMCluster, Get-WMCluster, Get-WMNamespace…}
Script     1.3.9                 VMware.vSphere.SsoAdmin             {Add-GroupToSsoGroup, Add-LDAPIdentitySource, Add-UserToSsoGroup, Connect-SsoAdminServer…}
Script     12.7.0.20…            VMware.VumAutomation                {Add-EntityBaseline, Copy-Patch, Get-Baseline, Get-Compliance…}

PS /home/vmware/.local/share/powershell/Modules> Import-Module -Name VMware.PowerCLI
PS /home/vmware/.local/share/powershell/Modules> Import-Module -Name VMware.vSphere.SsoAdmin
PS /home/vmware/.local/share/powershell/Modules> Import-Module -Name PowerVCF
PS /home/vmware/.local/share/powershell/Modules> Import-Module -Name PowerValidatedSolutions
PS /home/vmware/.local/share/powershell/Modules> Import-Module -Name VMware.CloudFoundation.PasswordManagement

PS /home/vmware/.local/share/powershell/Modules> Get-InstalledModule

Version              Name                                Repository           Description
-------              ----                                ----------           -----------
13.1.0.21605386      VMware.VimAutomation.Common         PSGallery            This PowerShell module contains functionality required by multiple PowerCLI modules.
8.1.0.21605554       VMware.Vim                          PSGallery            This PowerShell module contains PowerCLI Vim.
4.1.0.21605558       VMware.Sdk.Nsx.Policy               PSGallery            This PowerShell module contains cmdlets for managing VMware.Sdk.Nsx.Policy API.
13.1.0.21606089      VMware.VimAutomation.Nsxt           PSGallery            This PowerShell module contains PowerCLI NSXT cmdlets.
12.4.0.18627055      VMware.VimAutomation.WorkloadManag… PSGallery            PowerShell commands for automation of workloads related functionality in vSphere.
8.0.1111.21624264    VMware.Sdk.vSphere.ContentLibrary   PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.ContentLibra…
8.0.1111.21624264    VMware.Sdk.vSphere.Appliance.Infra… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Appliance.In…
8.0.1111.21624264    VMware.Sdk.vSphere.Appliance.Recov… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Appliance.Re…
13.1.0.21606510      VMware.VimAutomation.Security       PSGallery            This PowerShell module contains PowerCLI security management cmdlets.
12.6.0.19606210      VMware.CloudServices                PSGallery            PowerCLI CloudServices sample module.
12.7.0.20091290      VMware.VimAutomation.Srm            PSGallery            This Windows PowerShell module contains PowerCLI SRM cmdlets.
8.0.1111.21624264    VMware.Sdk.vSphere.Appliance.Suppo… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Appliance.Su…
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Namespa… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Name…
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Namespa… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Name…
8.0.1111.21624264    VMware.Sdk.vSphere.VAPI.Metadata    PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.VAPI.Metadat…
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.SystemC… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Syst…
8.0.1111.21624264    VMware.Sdk.vSphere.vStats           PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vStats.
13.1.0.21605170      VMware.VimAutomation.Sdk            PSGallery            This PowerShell module contains PowerCLI Sdk.
8.7.0.21605566       VMware.Sdk.Vr                       PSGallery            This PowerShell module contains cmdlets for managing VMware.Sdk.Vr API.
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Deploym… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Depl…
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Storage  PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Stor…
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Guest    PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Gues…
12.6.0.19600125      VMware.PowerCLI.VCenter             PSGallery            Modules for automating and managing VMware vCenter
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Authent… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Auth…
8.0.0.21610665       VMware.DeployAutomation             PSGallery            This PowerShell module contains PowerCLI Auto Deploy cmdlets.
8.0.1111.21624264    VMware.Sdk.vSphere.Appliance.Local… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Appliance.Lo…
12.6.0.19600125      VMware.PowerCLI.VCenter.Types.Cert… PSGallery            Types definitions for the VMware.PowerCLI.VCenter module related to the certificate management fe…
13.1.0.21605976      VMware.VimAutomation.Cis.Core       PSGallery            This PowerShell module contains PowerCLI Cloud Infrastructure Suite cmdlets.
8.0.1111.21624264    VMware.Sdk.vSphere.Appliance.Access PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Appliance.Ac…
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.VmTempl… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.VmTe…
8.0.1111.21624264    VMware.Sdk.vSphere.Appliance.Loggi… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Appliance.Lo…
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Vm       PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Vm.
2.3.0.1004           PowerVCF                            PSGallery            PowerShell Module for the VMware Cloud Foundation 4.x API
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Tagging  PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Tagg…
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Services PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Serv…
13.1.0.21610933      VMware.VimAutomation.Vds            PSGallery            This PowerShell module contains PowerCLI VDS cmdlets.
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.CertMan… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Cert…
1.0.1111.21624264    VMware.Sdk.Runtime                  PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.Runtime.
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Content  PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Cont…
1.2.0.1005           VMware.CloudFoundation.PasswordMan… PSGallery            PowerShell Module for VMware Cloud Foundation Password Management
1.6.0.0              VMware.VimAutomation.StorageUtility PSGallery            This Windows PowerShell module contains utility scripts for storage.
8.0.1111.21624264    VMware.Sdk.vSphere.Esx.Hosts        PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Esx.Hosts.
8.7.0.21605564       VMware.Sdk.Srm                      PSGallery            This PowerShell module contains cmdlets for managing VMware.Sdk.Srm API.
12.6.0.19600125      VMware.PowerCLI.Sdk                 PSGallery            Product agnostic types definitions for the VMware.PowerCLI.VCenter module.
8.0.1111.21624264    VMware.Sdk.vSphereRuntime           PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphereRuntime.
8.0.1111.21624264    VMware.Sdk.vSphere.Esx.Hcl          PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Esx.Hcl.
12.0.0.15939670      VMware.VimAutomation.License        PSGallery            This Windows PowerShell module contains PowerCLI cmdlets for managing VMware product licenses.
8.0.1111.21624264    VMware.Sdk.vSphere                  PSGallery            This PowerShell module is a parent module for VMware.Sdk.vSphere modules.
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Trusted… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Trus…
8.0.1111.21624264    VMware.Sdk.vSphere.Cis              PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Cis.
12.6.0.19600125      VMware.PowerCLI.VCenter.Types.Appl… PSGallery            Types definitions for the VMware.PowerCLI.VCenter module related to the appliance service managem…
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Identity PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Iden…
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter          PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.
8.0.1111.21624264    VMware.Sdk.vSphere.Esx.Settings     PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Esx.Settings.
8.0.0.21610262       VMware.ImageBuilder                 PSGallery            This PowerShell module contains PowerCLI ImageBuilder cmdlets.
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.ISO      PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.ISO.
2.4.0.1008           PowerValidatedSolutions             PSGallery            PowerShell Module for VMware Validated Solutions
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Topology PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Topo…
13.1.0.21606170      VMware.VimAutomation.Core           PSGallery            This PowerShell module contains Windows PowerShell cmdlets for managing vSphere.
13.0.0.20797723      VMware.VimAutomation.Vmc            PSGallery            This Windows PowerShell module contains PowerCLI VMC cmdlets.
8.0.1111.21624264    VMware.Sdk.vSphere.Appliance.Health PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Appliance.He…
13.1.0.21606282      VMware.VimAutomation.Storage        PSGallery            This Windows PowerShell module contains PowerCLI storage management cmdlets.
1.3.9                VMware.vSphere.SsoAdmin             PSGallery            PowerShell Module for Managing VMware vSphere SSO Admin functionality.
8.0.1111.21624264    VMware.Sdk.vSphere.Appliance.System PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Appliance.Sy…
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.OVF      PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.OVF.
8.0.1111.21624264    VMware.Sdk.vSphere.Appliance        PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Appliance.
13.1.0.21610272      VMware.VimAutomation.HorizonView    PSGallery            This PowerShell module contains Connect/Disconnect cmdlets for View API service.
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Authori… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Auth…
13.0.0.20803747      VMware.VimAutomation.Hcx            PSGallery            This Windows PowerShell module contains PowerCLI HCX cmdlets.
13.1.0.21624340      VMware.PowerCLI                     PSGallery            This Windows PowerShell module contains VMware.PowerCLI
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Consump… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Cons…
8.0.1111.21624264    VMware.Sdk.vSphere.Appliance.Netwo… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Appliance.Ne…
12.6.0.19600125      VMware.PowerCLI.Sdk.Types           PSGallery            Types definitions for the VMware.PowerCLI.Sdk module.
8.0.1111.21624264    VMware.Sdk.vSphere.Content          PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Content.
8.0.1111.21624264    VMware.Sdk.vSphere.Cis.Tagging      PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Cis.Tagging.
13.1.0.21611158      VMware.VimAutomation.vROps          PSGallery            This PowerShell module contains PowerCLI vROps cmdlets.
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Invento… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Inve…
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.Datasto… PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.Data…
12.7.0.20091294      VMware.VumAutomation                PSGallery            This Windows PowerShell module contains PowerCLI VUM cmdlets.
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.HVC      PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.HVC.
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.VCHA     PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.VCHA.
8.0.1111.21624264    VMware.Sdk.vSphere.vCenter.LCM      PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.vCenter.LCM.
8.0.1111.21624264    VMware.Sdk.vSphere.Appliance.Update PSGallery            This PowerShell module contains PowerShell Advanced functions for VMware.Sdk.vSphere.Appliance.Up…
13.1.0.21611174      VMware.VimAutomation.Cloud          PSGallery            This PowerShell module contains PowerCLI Cloud cmdlets.

PS /home/vmware/.local/share/powershell/Modules> Test-VcfPasswordManagementPrereq
[06-28-2023_19:11:04] [INFO] PowerShell Module: VMware.PowerCLI 13.1.0.21624340 is installed and supports the minimum required version.
[06-28-2023_19:11:05] [INFO] PowerShell Module: VMware.vSphere.SsoAdmin 1.3.9 is installed and supports the minimum required version.
[06-28-2023_19:11:07] [INFO] PowerShell Module: PowerVCF 2.3.0.1004 is installed and supports the minimum required version.
[06-28-2023_19:11:08] [INFO] PowerShell Module: PowerValidatedSolutions 2.4.0.1008 is installed and supports the minimum required version.
```
</details> 

<details><summary>But, as mentioned, unsuccessful on Windows Server 2019 with PowerShell 7.3.4</summary>

```powershell
PS C:\> cd 'C:\Program Files\WindowsPowerShell\Modules\'
PS C:\Program Files\WindowsPowerShell\Modules> Save-Module -Name VMware.PowerCLI -Repository PSGallery -Path .
PS C:\Program Files\WindowsPowerShell\Modules> Save-Module -Name VMware.vSphere.SsoAdmin -Repository PSGallery -Path .
PS C:\Program Files\WindowsPowerShell\Modules> Save-Module -Name PowerValidatedSolutions -Repository PSGallery -Path .
PS C:\Program Files\WindowsPowerShell\Modules> Save-Module -Name PowerVCF -Repository PSGallery -Path .
PS C:\Program Files\WindowsPowerShell\Modules> Save-Module -Name VMware.CloudFoundation.PasswordManagement -Repository PSGallery -Path .

PS C:\Program Files\WindowsPowerShell\Modules> Get-Module

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Binary     7.0.0.0               CimCmdlets                          {Get-CimAssociatedInstance, Get-CimClass, Get-CimInstance, Get-CimSession…}
Manifest   1.0.0.0               DnsClient                           {Resolve-DnsName, Add-DnsClientNrptRule, Clear-DnsClientCache, Get-DnsClient…}
Manifest   7.0.0.0               Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty…}
Manifest   7.0.0.0               Microsoft.PowerShell.Security       {ConvertFrom-SecureString, ConvertTo-SecureString, Get-Acl, Get-AuthenticodeSignature…}
Manifest   7.0.0.0               Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object…}
Manifest   7.0.0.0               Microsoft.WSMan.Management          {Connect-WSMan, Disable-WSManCredSSP, Disconnect-WSMan, Enable-WSManCredSSP…}
Manifest   2.0.0.0               NetAdapter                          {Disable-NetAdapter, Disable-NetAdapterBinding, Disable-NetAdapterChecksumOffload, Disable-NetAdapterEncapsulatedPacketTaskOffl…
Manifest   2.0.0.0               NetSecurity                         {Get-DAPolicyChange, New-NetIPsecAuthProposal, New-NetIPsecMainModeCryptoProposal, New-NetIPsecQuickModeCryptoProposal…}
Manifest   1.0.0.0               NetTCPIP                            {Find-NetRoute, Get-NetCompartment, Get-NetIPAddress, Get-NetIPConfiguration…}
Script     1.4.8.1               PackageManagement                   {Find-Package, Find-PackageProvider, Get-Package, Get-PackageProvider…}
Script     2.2.5                 PowerShellGet                       {Find-Command, Find-DscResource, Find-Module, Find-RoleCapability…}
Script     2.2.6                 PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSReadLineKeyHandler, Set-PSReadLineKeyHandler…}

PS C:\Program Files\WindowsPowerShell\Modules> Import-Module -Name VMware.PowerCLI
>> Import-Module -Name VMware.vSphere.SsoAdmin
>> Import-Module -Name PowerVCF
>> Import-Module -Name PowerValidatedSolutions
>> Import-Module -Name VMware.CloudFoundation.PasswordManagement
>> Import-Module -Name VMware.CloudFoundation.PasswordManagement
          Welcome to VMware PowerCLI!

Log in to a vCenter Server or ESX host:              Connect-VIServer
To find out what commands are available, type:       Get-VICommand
To show searchable help for all PowerCLI commands:   Get-PowerCLIHelp
Once you've connected, display all virtual machines: Get-VM
If you need more help, visit the PowerCLI community: Get-PowerCLICommunity

       Copyright (C) VMware, Inc. All rights reserved.

PS C:\Program Files\WindowsPowerShell\Modules> Get-InstalledModule

 [06-28-2023_21:11:25] [ERROR] PowerShell Module: VMware.PowerCLI 13.0.0 minimum required version is not installed.
PS C:\Program Files\WindowsPowerShell\Modules>
````
</details> 

Will provide an update soon on next steps, but we have planned to add the `RequiredModules` to the manifest to make this simpler.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,11 @@
 
 > Release Date: Unreleased
 
+Enhancement:
+
+- Added the `RequiredModules` key to the module manifest to specify the minimum dependencies required to install and run the PowerShell module. [GH-63](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/63)
+- Updated `Test-VcfPasswordManagementPrereq` to verify that the minimum dependencies are met to run the PowerShell module based on the module's manifest. [GH-63](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/63)
+
 Bug Fixes:
 
 - Updated `Request-SsoPasswordComplexity` to use `Test-VCFConnection` instead of `Test-Connection` to check the connection. [GH-62](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/62)

--- a/VMware.CloudFoundation.PasswordManagement.psd1
+++ b/VMware.CloudFoundation.PasswordManagement.psd1
@@ -12,7 +12,7 @@
     RootModule = '.\VMware.CloudFoundation.PasswordManagement.psm1'
     
     # Version number of this module.
-    ModuleVersion = '1.2.1.1000'
+    ModuleVersion = '1.2.1.1001'
     
     # Supported PSEditions
     # CompatiblePSEditions = @()
@@ -51,7 +51,24 @@
     # ProcessorArchitecture = ''
     
     # Modules that must be imported into the global environment prior to importing this module
-    # RequiredModules = @()
+    RequiredModules   = @(
+        @{
+            ModuleName    = 'VMware.PowerCLI'
+            ModuleVersion = '13.0.0'
+        }
+        @{
+            ModuleName    = 'VMware.vSphere.SsoAdmin'
+            ModuleVersion = '1.3.9'            
+        }
+        @{
+            ModuleName    = 'PowerVCF'
+            ModuleVersion = '2.3.0'
+        }
+        @{
+            ModuleName    = 'PowerValidatedSolutions'
+            ModuleVersion = '2.5.0'
+        }
+    )
     
     # Assemblies that must be loaded prior to importing this module
     # RequiredAssemblies = @()

--- a/VMware.CloudFoundation.PasswordManagement.psm1
+++ b/VMware.CloudFoundation.PasswordManagement.psm1
@@ -8524,42 +8524,54 @@ Export-ModuleMember -Function Update-LocalUserPasswordExpiration
 ##########################################################################
 #Region     Begin Supporting Functions                              ######
 
+
 Function Test-VcfPasswordManagementPrereq {
     <#
 		.SYNOPSIS
-        Validate prerequisites to run the PowerShell module.
+        Verifies that the minimum dependencies are met to run the PowerShell module.
 
         .DESCRIPTION
-        The Test-VcfPasswordManagementPrereq cmdlet checks that all the prerequisites have been met to run the PowerShell module.
+        The Test-VcfPasswordManagementPrereq cmdlet verifies that the minimum dependencies are met to
+        run the the PowerShell module.
 
         .EXAMPLE
         Test-VcfPasswordManagementPrereq
-        This example runs the prerequisite validation.
+        This example shows how to verify that the minimum dependencies are met to run the PowerShell module.
     #>
 
     Try {
-        Clear-Host; Write-Host ""
+        $moduleName = $myInvocation.myCommand.ModuleName
+        $moduleData = (Get-Module -Name $moduleName)
 
-        $modules = @(
-            @{ Name=("VMware.PowerCLI"); MinimumVersion=("13.0.0")}
-            @{ Name=("VMware.vSphere.SsoAdmin"); MinimumVersion=("1.3.9")}
-            @{ Name=("PowerVCF"); MinimumVersion=("2.3.0")}
-            @{ Name=("PowerValidatedSolutions"); MinimumVersion=("2.4.0")}
-        )
+        if ($PSEdition -eq 'Core' -and ($PSVersionTable.OS).Split(' ')[0] -eq 'Linux') {
+            $moduleManifestPath = $moduleData.ModuleBase + '/' + $moduleData.Name + '.psd1'
+        } else {
+            $moduleManifestPath = $moduleData.ModuleBase + '\' + $moduleData.Name + '.psd1'
+        }
 
-        foreach ($module in $modules ) {
-            if ((Get-InstalledModule -ErrorAction SilentlyContinue -Name $module.Name).Version -lt $module.MinimumVersion) {
-                $message = "PowerShell Module: $($module.Name) $($module.MinimumVersion) minimum required version is not installed."
-                Show-PasswordManagementOutput -type ERROR -message $message
-                Break
+        $moduleManifest = Import-PowerShellDataFile -Path $moduleManifestPath
+        $requiredModules = $moduleManifest.RequiredModules
+
+        foreach ($module in $requiredModules) {
+            $moduleName = $module.ModuleName
+            $requiredVersion = $module.ModuleVersion
+            $installedModule = Get-Module -ListAvailable -Name $moduleName
+
+            if ($installedModule) {
+                $installedVersion = $installedModule.Version
+                if ($installedVersion -lt $requiredVersion) {
+                    $message = "$($moduleName) $($installedVersion) is installed. Install $($moduleName) $($requiredVersion) or higher."
+                    Show-PasswordManagementOutput -type ERROR -message $message
+                } elseif ($installedVersion -ge $requiredVersion) {
+                    $message = "$($moduleName) $($installedVersion) is installed version and meets the minimum required version of $($moduleName) $($requiredVersion)."
+                    Show-PasswordManagementOutput -type INFO -message $message
+                }
             } else {
-                $moduleCurrentVersion = (Get-InstalledModule -Name $module.Name).Version
-                $message = "PowerShell Module: $($module.Name) $($moduleCurrentVersion) is installed and supports the minimum required version."
-                Show-PasswordManagementOutput -type INFO -message $message
+                $message = "$($moduleName) is not installed. Install $($moduleName) $($requiredVersion) or higher."
+                Show-PasswordManagementOutput -type ERROR -message $message
             }
         }
-    }
-    Catch {
+    } Catch {
         Write-Error $_.Exception.Message
     }
 }

